### ISSUE TITLE
mingw-w64-x86_64-SDL2_mixer can not be installed due to a signature issue

### ISSUE THREAD

**Issue by etavardt** [msys2/MINGW-packages#10988]
I am not able to install package mingw-w64-x86_64-SDL2_mixer due to a signature issue with ...

error: mingw-w64-x86_64-libvorbis: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-libvorbis-1.3.7-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).

I even tried to install the libvorbis package separately and get the same error.

**Comment by Biswa96** [comment]
Packaging keys were last updated on 29th Dec 2021. Did you miss any [msys2-keyring][1] update? Make sure everything is up-to-date with `pacman -Syyu` command. 

[1]: https://packages.msys2.org/base/msys2-keyring

**Comment by etavardt** [comment]
I had done this command in the last month or so, but just reran it again with the same result.

 pacman -Syyu
:: Synchronizing package databases...
 mingw32                                                    1480.8 KiB   664 KiB/s 00:02 [##################################################] 100%
 mingw64                                                    1487.8 KiB   541 KiB/s 00:03 [##################################################] 100%
 ucrt64                                                     1606.0 KiB   417 KiB/s 00:04 [##################################################] 100%
 clang64                                                    1525.7 KiB   870 KiB/s 00:02 [##################################################] 100%
 msys                                                        379.2 KiB   137 KiB/s 00:03 [##################################################] 100%
:: Starting core system upgrade...
 there is nothing to do
:: Starting full system upgrade...
 there is nothing to do

etava@MSI MSYS ~
 pacman -S mingw-w64-x86_64-SDL2_mixer
resolving dependencies...
looking for conflicting packages...

Packages (11) mingw-w64-x86_64-flac-1.3.4-1  mingw-w64-x86_64-fluidsynth-2.2.5-1  mingw-w64-x86_64-libmodplug-0.8.9.0-4
              mingw-w64-x86_64-libogg-1.3.5-1  mingw-w64-x86_64-libsndfile-1.0.31-4  mingw-w64-x86_64-libvorbis-1.3.7-1
              mingw-w64-x86_64-mpg123-1.29.3-2  mingw-w64-x86_64-opus-1.3.1-4  mingw-w64-x86_64-opusfile-0.12-2
              mingw-w64-x86_64-portaudio-1~19.7.0-4  mingw-w64-x86_64-SDL2_mixer-2.0.4-7

Total Download Size:    1.39 MiB
Total Installed Size:  15.11 MiB

:: Proceed with installation? [Y/n] Y
:: Retrieving packages...
 mingw-w64-x86_64-libsndfile-1.0.31-4-any                    459.6 KiB   330 KiB/s 00:01 [##################################################] 100%
 mingw-w64-x86_64-flac-1.3.4-1-any                           510.2 KiB   358 KiB/s 00:01 [##################################################] 100%
 mingw-w64-x86_64-libvorbis-1.3.7-1-any                      360.3 KiB   249 KiB/s 00:01 [##################################################] 100%
 mingw-w64-x86_64-SDL2_mixer-2.0.4-7-any                      96.9 KiB  47.3 KiB/s 00:02 [##################################################] 100%
 Total (4/4)                                                1427.0 KiB   591 KiB/s 00:02 [##################################################] 100%
(11/11) checking keys in keyring                                                         [##################################################] 100%
(11/11) checking package integrity                                                       [##################################################] 100%
error: mingw-w64-x86_64-libvorbis: signature from "David Macek [<david.macek.0@gmail.com>](mailto:david.macek.0@gmail.com)" is unknown trust
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-libvorbis-1.3.7-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] Y
error: failed to commit transaction (invalid or corrupted package)
Errors occurred, no packages were upgraded.

**Comment by Biswa96** [comment]
There are some workarounds in this thread https://github.com/msys2/MSYS2-packages/issues/2343. Theoretically, this issue should not occur if packages are updated regularly.

**Comment by etavardt** [comment]
Does pacman -Qs msys2-keyring command show like this?

$ pacman -Qs msys2-keyring
local/msys2-keyring 1~20211228-1
    MSYS2 PGP keyring

Also do these command help? _**Answer: Yes**_

rm -rf /etc/pacman.d/gnupg
pacman-key --init
pacman -S msys2-keyring
pacman-key --populate msys2
pacman -Syu

Running these commands appeared to make things worse and appeared to fail at repairing the gnupg or keyring.
After trying several times I decided to run pacman -Syuu again but that appeared to have issues as well now.
I then deceided to close/restart the shell/msys window, and some kind of repair appeared to take place.  It did show some errors when attempting to do some email checks.  When it completed it said I needed to restart the window so I did and reran the pacman -Syuu again and all appeared to be up to date and fine.  I then tried to install the mixer again and everything installed.  So all is good now.  I do not know what exactly caused it do finally work.

**Comment by etavardt** [comment]
Does pacman -Qs msys2-keyring command show like this?

$ pacman -Qs msys2-keyring
local/msys2-keyring 1~20211228-1
MSYS2 PGP keyring

Also do these command help?/*Answer: Yes*/

rm -rf /etc/pacman.d/gnupg
pacman-key --init
pacman -S msys2-keyring
pacman-key --populate msys2
pacman -Syu

Running these commands appeared to make things worse and appeared to 
fail at repairing the gnupg or keyring.
After trying several times I decided to run pacman -Syuu again but that 
appeared to have issues as well now.
I then deceided to close/restart the shell/msys window, and some kind of 
repair appeared to take place. It did show some errors when attempting 
to do some email checks. When it completed it said I needed to restart 
the window so I did and reran the pacman -Syuu again and all appeared to 
be up to date and fine. I then tried to install the mixer again and 
everything installed. So all is good now. I do not know what exactly 
caused it do finally work.


Thank you.
